File: data_structures_algorithms/LinkedLists/python/singlyLL.py
class Node:

	def __init__(self, data):

		self.data = data
		self.next = None

class LinkedList:

	def __init__(self):

		self.head = None

	def push(self, data):

		tmp = Node(data)
		tmp.next = self.head
		self.head = tmp;

	def insert_after(self, prev_node, data):

		if prev_node is None:
			return -1

		tmp = Node(data)
		tmp.next = prev_node.next
		prev_node.next = tmp;

	def append(self, data):

		tmp = Node(data)

		if self.head is None:
			self.head = tmp
			return 0

		last = self.head
		while (last.next):
			last = last.next

		last.next = tmp

	def print_list(self):

		tmp = self.head

		while(tmp):
			print(tmp.data)
			tmp = tmp.next

File: data_structures_algorithms/LinkedLists/c/singlyLL.c
struct Node {

	int data;
	struct Node *next;
};

int push(struct Node **head, int data) {

	struct Node *tmp = (struct Node*)malloc(sizeof(struct Node));
	tmp->data = data;
	tmp->next = (*head);
	(*head) = tmp;
}

int insertAfter(struct Node *prevNode, int data) {

	if (prevNode == NULL) {
		return -1;
	}

	struct Node *tmp = (struct Node *)malloc(sizeof(struct Node));
	tmp->data = data;
	tmp->next = prevNode->next;
	prevNode->next = tmp;

}

void append(struct Node **head, int data) {

	struct Node *tmp = (struct Node *)malloc(sizeof(struct Node));

	struct Node *last = *head;

	tmp->data = data;
	tmp->next = NULL;

	if (*head == NULL) {
		*head = tmp;
		return;
	}

	while (last->next != NULL) {
		last = last->next;
	}

	last->next = tmp;

	return;
}

void printList(struct Node *node) {

	while(node != NULL) {

		printf("%d ", node->data);
		node = node->next;
	}
}